Podcast/SearchViewController.swift

```diff
@@ -155,5 +386,95 @@ class SearchViewController: ViewController, UISearchControllerDelegate, UITableV
         UserDefaults.standard.set([], forKey: "PastSearches")
         pastSearchesTableViewReloadData()
     }
-    
+}
+
+protocol SearchTableViewDelegate: class {
+    func didSelectCell(cell: UITableViewCell, object: Any)
+    func didPressFollowButton(cell: SearchPeopleTableViewCell)
+    func didPressSubscribeButton(cell: SearchSeriesTableViewCell)
+    func didPressPlayButton(cell: SearchEpisodeTableViewCell)
+    func refreshController()
+}
+
+class SearchDataSourceDelegate: NSObject, UITableViewDataSource, UITableViewDelegate, SearchEpisodeTableViewCellDelegate, SearchSeriesTableViewDelegate, SearchPeopleTableViewCellDelegate {
+
+    var searchResults: [Any] = []
+    var searchType: SearchType
+    var pageSize: Int = 20
+    var continueInfiniteScroll: Bool = true
+    var completingNewSearch: Bool = false // two types of searching: inifinite scroll or new search query -> true if a new search query in progress
+
+    weak var delegate: SearchTableViewDelegate?
+
+    init(searchType: SearchType) {
+        self.searchType = searchType
+    }
+
+    func fetchData(query: String, newSearch: Bool = false) {
+        if newSearch { continueInfiniteScroll = true }
+        System.endpointRequestQueue.cancelAllEndpointRequestsOfType(type: searchType.endpointType)
+        let request = searchType.endpointType.init(modelPath: searchType.path, query: query, offset: newSearch ? 0 : searchResults.count, max: pageSize)
+        completingNewSearch = newSearch
+        request.success = { endpoint in
+            guard let results = endpoint.processedResponseValue as? [Any] else { return }
+            self.completingNewSearch = false
+            if results.count < self.pageSize { self.continueInfiniteScroll = false }
+            if newSearch {
+                self.searchResults = results
+            } else {
+                self.searchResults = self.searchResults + results
+            }
+            self.delegate?.refreshController()
```

Comment:
![image](https://user-images.githubusercontent.com/20246620/34462896-48f06370-ee1b-11e7-9f1f-102da4e7400f.png)
should the refresh controller be consistent with the other loading indicators (the cool one that's used elsewhere)

Reply:
We've been using this one for refreshing and infinite scroll ... @dantheli and I looked into making it the same indicator and it didn't work / neither of us could figure out why `NVAcitvityIndicatorView` did not show up for an infinite scroll. I the default one looks fine so let's keep it for now.